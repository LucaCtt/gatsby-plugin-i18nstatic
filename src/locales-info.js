export const LocalesInfoContext = createContext({
  currentLocale: "",
  defaultLocale: "",
  locales: []
});

export const useLocalesInfo = () => {
  const { currentLocale, defaultLocale, locales } = useContext(
    LocalesInfoContext
  );
  return [currentLocale, defaultLocale, locales];
};
