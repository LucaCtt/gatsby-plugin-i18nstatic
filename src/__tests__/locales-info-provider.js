import React from "react";
import { render } from "@testing-library/react";

import LocalesInfoProvider from "../locales-info-provider";
import { useLocalesInfo } from "../locales-info";

const MOCK_CURRENT_LOCALE = "it";
const MOCK_DEFAULT_LOCALE = "en";
const MOCK_LOCALES = ["en", "it", "es"];

const TestComponent = ({ localesInfo, children, ...props }) => (
  <LocalesInfoProvider localesInfo={localesInfo} {...props}>
    {children}
  </LocalesInfoProvider>
);

const Consumer = () => {
  const [currentLocale, defaultLocale, locales] = useLocalesInfo();

  return (
    <div>
      <p>
        {currentLocale}
        {defaultLocale}
      </p>
      <ul>
        {locales.map(l => (
          <li key={l}>{l}</li>
        ))}
      </ul>
    </div>
  );
};

test("Renders correctly", () => {
  const { container } = render(
    <TestComponent
      localesInfo={{
        currentLocale: MOCK_CURRENT_LOCALE,
        defaultLocale: MOCK_DEFAULT_LOCALE,
        locales: MOCK_LOCALES
      }}
    >
      <Consumer />
    </TestComponent>
  );
  expect(container.firstChild).toMatchSnapshot();
});
