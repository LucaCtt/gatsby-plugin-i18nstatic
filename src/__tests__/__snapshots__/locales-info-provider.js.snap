// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <p>
    it
    en
  </p>
  <ul>
    <li>
      en
    </li>
    <li>
      it
    </li>
    <li>
      es
    </li>
  </ul>
</div>
`;
